lib/dust.js

```diff
@@ -792,14 +804,9 @@
       if (body) {
         len = elem.length;
         if (len > 0) {
-          head = context.stack && context.stack.head || {};
-          head.$len = len;
           for (i = 0; i < len; i++) {
-            head.$idx = i;
             chunk = body(chunk, context.push(elem[i], i, len));
```

Comment:
Speaking of smell. The params for idx and len come from here, get funneled through `context.push` just to be stored on `stack`.